src/CssInlinerFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier;
+
+use DOMDocument;
+
+/**
+ * This class provides a factory for CssInliner that can be passed as a dependency
+ * or be injected via dependency injection
+ *
+ * @author SpazzMarticus <SpazzMarticus@users.noreply.github.com>
+ */
+class CssInlinerFactory
+{
+    /**
+     * Creates a new instance of CssInliner from a string
+     *
+     * @param string $unprocessedHtml
+     *
+     * @return CssInliner
+     *
+     * @throws \InvalidArgumentException if $unprocessedHtml is anything other than a non-empty string
+     */
+    public function createFromHtml(string $unprocessedHtml): CssInliner
+    {
+        return CssInliner::fromHtml($unprocessedHtml);
+    }
+
+    /**
+     * Creates a new instance of CssInliner from a DomDocument
+     *
+     * @param DOMDocument $document
+     *
+     * @return CssInliner
+     */
+    public function createFromDomDocument(DOMDocument $document): CssInliner
```

Comment:
Do we need this? Usually, `fromDomDocument` should not be called by users, but only internally (or with data from other `HtmlProcessor` classes).

Reply:
Since the [more complex example](https://github.com/MyIntervals/emogrifier#more-complex-example) chains together different classes using `fromDomDocument` I would keep it in.